subprocess_linux.go

```diff
@@ -0,0 +1,235 @@
+}
+
+// LinuxWeightDevice represents the weighting for time-based division of disk policy.
+type LinuxWeightDevice struct {
+	// Weight is the bandwidth rate for the device.
+	Weight *uint16 `json:"weight,omitempty"`
+	// LeafWeight is the bandwidth rate for the device while competing with the cgroup's child cgroups, CFQ scheduler only
+	LeafWeight *uint16 `json:"leafWeight,omitempty"`
+}
+
+// LinuxThrottleDevice represents the upper limit on I/O rate to a device.
+type LinuxThrottleDevice struct {
+	// Rate is the IO rate limit per cgroup per device
+	Rate uint64 `json:"rate"`
+}
+
+// LinuxHugepageLimit represents limits for the hugetlb subsystem.
+type LinuxHugepageLimit struct {
+	// Pagesize is the hugepage size
+	Pagesize string `json:"pageSize"`
+	// Limit is the limit of "hugepagesize" hugetlb usage
+	Limit uint64 `json:"limit"`
+}
+
+// LinuxNetwork represents priorities for the net_cls,net_prio subsystems.
+type LinuxNetwork struct {
+	// Set class identifier for container's network packets
+	ClassID *uint32 `json:"classID,omitempty"`
+	// Set priority of network traffic for container
+	Priorities []LinuxInterfacePriority `json:"priorities,omitempty"`
+}
+
+// LinuxInterfacePriority represents a priority on a network interface.
+type LinuxInterfacePriority struct {
+	// Name is the name of the network interface
+	Name string `json:"name"`
+	// Priority for the interface
+	Priority uint32 `json:"priority"`
+}
+
+// LinuxRdma represents limits on the rdma subsystem.
+type LinuxRdma struct {
+	// Maximum number of HCA handles that can be opened. Default is "no limit".
+	HcaHandles *uint32 `json:"hcaHandles,omitempty"`
+	// Maximum number of HCA objects that can be created. Default is "no limit".
+	HcaObjects *uint32 `json:"hcaObjects,omitempty"`
+}
+
+// getActivePIDs returns a list of active PIDs on the system by listing the
+// contents of /proc and looking for entries that appear to be valid PIDs.
+func getActivePIDs() ([]int, error) {
+	d, err := os.Open("/proc")
+	if err != nil {
+		return nil, err
+	}
+	defer d.Close()
+
+	results := make([]int, 0, 50)
+	for {
+		fileInfos, err := d.Readdir(10)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		for _, fileInfo := range fileInfos {
+			// PID must be a directory with a numeric name.
+			if !fileInfo.IsDir() {
+				continue
+			}
+
+			// Using Atoi here will also filter out "." and "..".
+			pid, err := strconv.Atoi(fileInfo.Name())
+			if err != nil {
+				continue
+			}
+			results = append(results, pid)
+		}
+	}
+	return results, nil
+}
+
+// getEnvironmentVariables returns a map of environment variables for the
+// given pid. This function works by reading from /proc/${pid}/environ, so the
+// entries returned only reflect the values of the environment variables at the
+// time that the process was started.
+func getEnvironmentVariables(pid int) (map[string]string, error) {
```

Comment:
same, can't we look at the processes in a manager to find this out?

Reply:
Removed this.